src/main/java/net/mcreator/blockly/BlocklyBlockUtil.java

```diff
@@ -118,4 +124,73 @@ public static Color getBlockColorFromHUE(int hue) {
 		return Color.getHSBColor(hue / 360f, 0.37f, 0.6f);
 	}
 
+	/**
+	 * Calculate the HUE of a block from RGB
+	 *
+	 * @param varElementType The variable element to use
+	 * @return Built-in color or HUE for other blocks
+	 */
+	public static String getBlocklyColorFromVarType(VariableElementType varElementType) {
+		//Built-in colors
+		if (varElementType.getName().equalsIgnoreCase("string"))
+			return "\"%{BKY_TEXTS_HUE}\"";
+		else if (varElementType.getName().equalsIgnoreCase("number"))
+			return "\"%{BKY_MATH_HUE}\"";
+		else if (varElementType.getName().equalsIgnoreCase("logic"))
+			return "\"%{BKY_LOGIC_HUE}\"";
+
+		//Custom colors
+		float[] hsbValues = new float[3];

```

Comment:
I think you can merge this into one line and put null as argument according to https://docs.oracle.com/javase/7/docs/api/java/awt/Color.html#RGBtoHSB(int,%20int,%20int,%20float[])